com.teamscale.polarion.plugin/src/com/teamscale/polarion/plugin/WorkItemUpdatesServlet.java

```diff
@@ -0,0 +1,769 @@
+package com.teamscale.polarion.plugin;
+
+import com.google.gson.Gson;
+import com.polarion.alm.projects.model.IProject;
+import com.polarion.alm.tracker.ITrackerService;
+import com.polarion.alm.tracker.model.ILinkedWorkItemStruct;
+import com.polarion.alm.tracker.model.IModule;
+import com.polarion.alm.tracker.model.IWorkItem;
+import com.polarion.core.util.logging.ILogger;
+import com.polarion.core.util.logging.Logger;
+import com.polarion.platform.core.PlatformContext;
+import com.polarion.platform.persistence.IDataService;
+import com.polarion.platform.persistence.UnresolvableObjectException;
+import com.polarion.platform.persistence.diff.IDiffManager;
+import com.polarion.platform.persistence.diff.IFieldDiff;
+import com.polarion.platform.persistence.model.IPObject;
+import com.polarion.platform.persistence.model.IPObjectList;
+import com.polarion.platform.security.PermissionDeniedException;
+import com.polarion.platform.service.repository.AccessDeniedException;
+import com.teamscale.polarion.plugin.model.LinkBundle;
+import com.teamscale.polarion.plugin.model.LinkFieldDiff;
+import com.teamscale.polarion.plugin.model.LinkedWorkItem;
+import com.teamscale.polarion.plugin.model.Response;
+import com.teamscale.polarion.plugin.model.WorkItemChange;
+import com.teamscale.polarion.plugin.model.WorkItemFieldDiff;
+import com.teamscale.polarion.plugin.model.WorkItemForJson;
+import com.teamscale.polarion.plugin.utils.Utils;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * This is the servlet that represents the endpoint for the Teamscale Polarion plugin. Its main job
+ * is to return a json object representing updates on work items of a particular document, in a
+ * given folder (space) and project.
+ *
+ * @author Bruno da Silva
+ */
+public class WorkItemUpdatesServlet extends HttpServlet {
```

Comment:
This class is pretty large. Could you please split it up into multiple classes?

Reply:
Will do that.

Reply:
Done